### PR TITLE
Update tornado_xstatic.py

### PR DESCRIPTION
**Pull request by liudonghua123**
try to fix some problems on windows platform. Correctly set path parameter, or you will get errors on https://github.com/tornadoweb/tornado/blob/v6.1.0/tornado/web.py#L2768.

### CODE REVIEW

**Comment by takluyver** [comment]
I'm amazed that a) someone is still using this, and b) it still even remotely works. I haven't touched it for 7 years! :laughing: 

Thanks for figuring this out, but I don't think the fix is quite right. You're assuming that any Xstatic data packages are installed next to tornado_xstatic in the same directory, which will often be true but not always. Given that most web applications are run on Linux <sup>[citation needed] :wink: </sup>, I don't want a fix for Windows that makes things worse on Linux and all other unix-y systems.

What could you do instead? Well, if your change works for you, you could just continue using it without me merging it. I'm not likely to make any 'upstream' changes for you to worry about. Otherwise, I think we'd need to override more of the StaticFileHandler methods to bypass the 'is this file within this directory' check - setting the directory to `/` is just a convenient shortcut for that on unix-y systems.

**Comment by liudonghua123** [comment]
Yes, I agree with you, this fix is for windows and it maybe break on none windows. How about setting the `path` according to the OS (whether windows or not) ?

```python
import sys
'/' if sys.platform != 'win32' else os.path.join(os.path.dirname(__file__), 'xstatic/pkg')
```

or this will works too (preferred, use the root driver path of the partition, more effective on https://github.com/tornadoweb/tornado/blob/v6.1.0/tornado/web.py#L2767)

```python
from pathlib import Path
'/' if sys.platform != 'win32' else Path(__file__).parts[0]
```

**Comment by takluyver** [comment]
I don't think either of them are 100% correct, because you could have files installed on different drives in Windows. But it's probably good enough 99% of the time, if you want to do that with a comment explaining it. `Path` objects also have a `.drive` attribute if you want to make it more explicit.

**Comment by liudonghua123** [comment]
I updated the code, the path is the root driver path of `xstatic` package now. This should always work. I could not find other situation which will not work. 
@takluyver any ideas about this?

File: tornado_xstatic.py [hunk 1]
```diff
@@ -14,10 +14,16 @@ def initialize(self, allowed_modules=None, **kwargs):
...
         assert 'root' not in kwargs
-        # NOTE: Not wild on passing path=/ , because StaticFileHandler's own
-        # validation will let this serve any file. If this subclass is working
-        # correctly, that shouldn't be an issue, but...
-        super(XStaticFileHandler, self).initialize(path="/")
+        # NOTE: Passing path=/ will not work on some conditions on windows
+        # like the pip packages are install in C:/ and the code you are
+        # running is in another partition.
+        # see also https://github.com/tornadoweb/tornado/blob/v6.1.0/tornado/web.py#L2768
+        # So change this to the root driver location of 'xstatic' package
```
Comment (takluyver) [inline tornado_xstatic.py]:
```suggestion
        # So change this to the root drive location of 'xstatic' package
```

**Comment by takluyver** [comment]
I think it's possible to engineer a situation where it won't work, because Xstatic is a namespace package, so you could install e.g. Xstatic-jquery somewhere in C: and Xstatic-termjs somewhere in D:. It's hard to come up with a realistic scenario where that would happen, though, so I think this is good enough.

**Comment by takluyver** [comment]
I imagine you'll want a new release of this now? Or do you think there's anything else to change before I do that?

**Comment by liudonghua123** [comment]
A new release will be perfect. 😄 
Currently, this package works well for me. I do not have anything to change.

**Comment by takluyver** [comment]
Great. It's a credit to @bdarnell that years worth of changes in Tornado - 4.1 to 6.1 - haven't broken this.

**Comment by takluyver** [comment]
0.3 is out now.

**Pull request merged**